parser/parser_test.go

```diff
@@ -86,153 +66,153 @@ func TestAAAASetup(t *testing.T) {
+		test{&paramInput{";foo=bar;baz=", ';', ';', 0, false, false}, &paramResult{pass, base.Params{"foo": base.String{"bar"}, "baz": base.String{""}}, 13}},
 		test{&paramInput{"$foo=bar,baz=,a=b", '$', ',', 0, false, true}, &paramResult{pass,
-			map[string]*string{"foo": &bar, "baz": &empty, "a": &b}, 17}},
-		test{&paramInput{"$foo=bar,baz,a=b", '$', ',', 0, false, false}, &paramResult{fail, map[string]*string{}, 17}},
-		test{&paramInput{";foo=\"bar\"", ';', ';', 0, false, true}, &paramResult{pass, map[string]*string{"foo": &barQuote}, 10}},
-		test{&paramInput{";foo=\"bar", ';', ';', 0, false, true}, &paramResult{pass, map[string]*string{"foo": &barQuote2}, 9}},
-		test{&paramInput{";foo=bar\"", ';', ';', 0, false, true}, &paramResult{pass, map[string]*string{"foo": &barQuote3}, 9}},
-		test{&paramInput{";\"foo\"=bar", ';', ';', 0, false, true}, &paramResult{pass, map[string]*string{"\"foo\"": &bar}, 10}},
-		test{&paramInput{";foo\"=bar", ';', ';', 0, false, true}, &paramResult{pass, map[string]*string{"foo\"": &bar}, 9}},
-		test{&paramInput{";\"foo=bar", ';', ';', 0, false, true}, &paramResult{pass, map[string]*string{"\"foo": &bar}, 9}},
-		test{&paramInput{";foo=\"bar\"", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &bar}, 10}},
-		test{&paramInput{";foo=\"ba\"r", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";foo=ba\"r", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";foo=bar\"", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";foo=\"bar", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";\"foo\"=bar", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";\"foo=bar", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";foo\"=bar", ';', ';', 0, true, true}, &paramResult{fail, map[string]*string{}, 0}},
-		test{&paramInput{";foo=\"bar;baz\"", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &barBaz}, 14}},
-		test{&paramInput{";foo=\"bar;baz\";a=b", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &barBaz, "a": &b}, 18}},
-		test{&paramInput{";foo=\"bar;baz\";a", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &barBaz, "a": nil}, 16}},
-		test{&paramInput{";foo=bar", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &bar}, 8}},
-		test{&paramInput{";foo=", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &empty}, 5}},
-		test{&paramInput{";foo=\"\"", ';', ';', 0, true, true}, &paramResult{pass, map[string]*string{"foo": &empty}, 7}},
+			base.Params{"foo": base.String{"bar"}, "baz": base.String{""}, "a": base.String{"b"}}, 17}},
```

Comment:
I really hope you found a way to avoid doing all this by hand. I'm just going to assume you got these right :)